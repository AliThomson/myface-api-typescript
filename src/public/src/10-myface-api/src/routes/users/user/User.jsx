import React from "react";
import { Link } from "react-router-dom";
import './user.scss';

export function User( {user }) {
    return (
        <div className="user">
            <img className="cover-image"
                src={user.coverImageUrl}
                alt={user.name}
                // width="400"
                // height="400"
                />
            <Link className="user-name" to={`/users/${user.id}`}>
                <h2>{user.name}</h2>
                </Link>
            <img className="user-image"
                src={user.profileImageUrl}
                alt={user.name}
                />
            <div className="user-info">
                <p>{user.email}</p>

                {user.likes !== undefined
                    ? <p> Liked: {user.likes.length} </p>
                    : <p></p>}

                {user.dislikes !== undefined
                    ? <p> Disliked: {user.dislikes.length} </p>
                    : <p></p>}
            </div>
            
        </div>
    )
}