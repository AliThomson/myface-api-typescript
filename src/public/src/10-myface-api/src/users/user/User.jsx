import React from "react";
import './user.css';

export function User( {user }) {
    return (
        <div className="user">
            <img className="cover-image"
                src={user.coverImageUrl}
                alt={user.name}
                width="400"
                height="400"
                />
            <h2>{user.name}</h2>
            <img className="profile-image"
                src={user.profileImageUrl}
                alt={user.name}
                />
            <p>{user.email}</p>
            <p>Liked: {user.likes.length}</p>
            <p>Disliked: {user.dislikes.length}</p>
            
        </div>
    )
}