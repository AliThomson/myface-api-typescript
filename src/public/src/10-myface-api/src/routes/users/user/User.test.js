import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import { User } from "./User";

test("renders name", async () => {
  // Arrange
  const user = {
    name: "ALing",
  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>);
  const nameElement = screen.getByText(/ALing/i);
  
  // Assert
  expect(nameElement).toBeInTheDocument();
});

test("renders image url", async () => {
  // Arrange
  const user = {
    name: "ALing",
    profileImageUrl: "Profile Image url",
    email: "ling@softwire.com",
  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>);
  const imageElement = screen.getByText(/Profile Image url/i);
  
  // Assert
  expect(imageElement).toBeInTheDocument();
});

test("renders email", async () => {
  // Arrange
  const user = {
    email: "ling@softwire.com",
  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>);
  const emailElement = screen.getByText(/ling@softwire.com/i);

  // Assert
 expect(emailElement).toBeInTheDocument();
});

test("matches snapshot", async () => {
  // Arrange
  const user = {
    name: "ALing",
    profileImageUrl: "Profile Image url",
    email: "ling@softwire.com",
    };

  // Act
  const tree = renderer
    .create(<BrowserRouter><User user={user} /></BrowserRouter>)
    .toJSON();
    
  // Assert
  expect(tree).toMatchSnapshot();
});

