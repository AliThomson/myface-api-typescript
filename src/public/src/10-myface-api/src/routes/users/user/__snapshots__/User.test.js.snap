// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="user"
>
  <img
    alt="ALing"
    className="cover-image"
    height="400"
    width="400"
  />
  <a
    href="/users/undefined"
    onClick={[Function]}
  >
    <h2>
      ALing
    </h2>
  </a>
  <img
    alt="ALing"
    src="Profile Image url"
  />
  <div>
    <p>
      ling@softwire.com
    </p>
    <p />
    <p />
  </div>
</div>
`;
